File: android/common/src/main/java/com/halo/common/utils/Strings.java
/*
 * Copyright 10/10/2018 Hahalolo. All rights reserved.
 *
 * https://help.hahalolo.com/commercial_terms/
 */

package com.halo.common.utils;

import android.text.TextUtils;

import com.google.gson.Gson;
import com.halo.constant.BuildConfig;

/**
 * @author ndn
 * Created by ndn
 * Created on 6/11/18.
 */
public final class Strings {

    public static void log(Object o) {
        if (BuildConfig.DEBUG) log("", o);
    }

    public static void log(String message, Object o) {

        String json = "";
        if (o instanceof String) {
            json = (String) o;
        } else {
            json = (o == null ? "NULL" : new Gson().toJson(o));
        }
        int maxLogSize = 2600;
        int partSize = json.length() / maxLogSize;
        if (partSize > 0 && BuildConfig.DEBUG)
            android.util.Log.d("TESTAPP_ANHSONDAY", (TextUtils.isEmpty(message) ? "" : (message + ": ")) + "Size: " + json.length());

        for (int i = 0; i <= partSize; i++) {
            int start = i * maxLogSize;
            int end = (i + 1) * maxLogSize;
            end = end > json.length() ? json.length() : end;
            if (BuildConfig.DEBUG) android.util.Log.d("TESTAPP_ANHSONDAY",
                    (TextUtils.isEmpty(message)
                            ? "" : (message + ": "))
                            + (partSize > 0 ? "PART " + (i + 1) + ": " : "")
                            + json.substring(start, end));
        }
    }
}

File: android/messenger/src/main/java/com/hahalolo/messager/utils/Strings.kt
/*
 * Copyright 10/10/2018 Hahalolo. All rights reserved.
 *
 * https://help.hahalolo.com/commercial_terms/
 */
package com.hahalolo.messager.utils

import android.text.TextUtils
import android.util.Log
import com.google.gson.Gson
import java.text.Normalizer
import java.util.regex.Pattern

object Strings {
    fun deAccent(str: String?): String? {
        try {
            return str?.run {
                val nfdNormalizedString = Normalizer.normalize(str, Normalizer.Form.NFD)
                val pattern = Pattern.compile("\\p{InCombiningDiacriticalMarks}+")
                pattern.matcher(nfdNormalizedString).replaceAll("")
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return null
    }

    fun logPer(`object`: Any) {
        val elements = Thread.currentThread().stackTrace
        var method = ""
        method = elements[3].methodName
        val content = `object`.javaClass.simpleName + "::" + method + "():"
        log(content, "")
    }

    fun logPer(`object`: Any, o: Any?) {
        val elements = Thread.currentThread().stackTrace
        var method = ""
        method = elements[3].methodName
        val content = `object`.javaClass.simpleName + "::" + method + "():"
        log(content, o)
    }

    fun log(o: Any?) {
        log("", o)
    }

    fun log(message: String, o: Any?) {
        var json = ""
        json = if (o is String) {
            o
        } else {
            if (o == null) "NULL" else Gson().toJson(o)
        }
        val maxLogSize = 2600
        val partSize = json.length / maxLogSize
        if (partSize > 0) Log.d(
            "TESTAPP_ANHSONDAY",
            (if (TextUtils.isEmpty(message)) "" else "$message: ") + "Size: " + json.length
        )
        for (i in 0..partSize) {
            val start = i * maxLogSize
            var end = (i + 1) * maxLogSize
            end = Math.min(end, json.length)
            Log.d(
                "TESTAPP_ANHSONDAY",
                (if (TextUtils.isEmpty(message)) "" else "$message: ")
                        + (if (partSize > 0) "PART " + (i + 1) + ": " else "")
                        + json.substring(start, end)
            )
        }
    }
}